plugins/android/src/main/java/tools/forma/android/plugin/PluginWrapper.kt

```diff
@@ -0,0 +1,38 @@
+package tools.forma.android.plugin
+
+import tools.forma.deps.FormaDependency
+import emptyDependency
+import org.gradle.api.Project
+import org.gradle.kotlin.dsl.apply
+import org.gradle.kotlin.dsl.the
+import tools.forma.deps.applyDependencies
+import tools.forma.validation.EmptyValidator
+import Forma
+import kotlin.reflect.KClass
+
+open class PluginWrapper<TPluginExtension : Any>(
+        protected val pluginId: String,
+        protected val pluginExtension: KClass<TPluginExtension>? = null,
+        protected val dependencies: FormaDependency = emptyDependency(),
+        protected val pluginConfiguration: (TPluginExtension.() -> Unit)? = null
+) {
+
+    init {
+        if (pluginConfiguration != null) {
```

Comment:
instead of doing this check, we can create a data class that will contain plugin extension and plugin configuration so we can reuse it later if, for example, different targets would need a slightly different configuration. This way we make sure Plugin Extension and Plugin configuration will always be specified together + compile-time check instead of runtime.

Reply:
yep, nice idea